deb/build-deb

```diff
@@ -40,8 +40,11 @@ debSource="$(awk -F ': ' '$1 == "Source" { print $2; exit }' debian/control)"
 debMaintainer="$(awk -F ': ' '$1 == "Maintainer" { print $2; exit }' debian/control)"
 debDate="$(date --rfc-2822)"
 
+# Note: on the 20.10 branch, we preserve the "~3" suffix that was removed in
+#       commit 8a720b28a5d0db8fc93423cca79ca21b9b69f1f1, and continue using the
+#       old version scheme for docker-ce, docker-ce-cli, and docker-rootless-extras.
 cat > "debian/changelog" <<-EOF
-$debSource (${EPOCH}${EPOCH_SEP}${DEB_VERSION}-0~${DISTRO}-${SUITE}) $SUITE; urgency=low
+$debSource (${EPOCH}${EPOCH_SEP}${DEB_VERSION}~3-0~${DISTRO}-${SUITE}) $SUITE; urgency=low
```

Comment:
Revert the changes from 8a720b28a5d0db8fc93423cca79ca21b9b69f1f1 for the docker-ce packages (but continue using the new format for the plugins, so that we don't push the same version of a plugin twice (once through the 23.0 release, once through the 20.10 release).

Reply:
Wait, what's the `~3` here for?  (Sorry, I'm missing some context :see_no_evil:)

Reply:
This was used in 20.10 (and before) to.. make sure "stable" releases ranked higher than "pre-releases" (`:lolsob:`). I removed that code from master, but the same generation code is used for all deb versions, so I backported removing that ugly hack in this PR (see https://github.com/docker/docker-ce-packaging/pull/831/commits/8a720b28a5d0db8fc93423cca79ca21b9b69f1f1). However, as 20.10 is reaching its end, I didn't want to introduce changing the versioning scheme for it, so I added it back here. As we're not planning to do `-beta` releases for 20.10 patch releases, I hardcoded it to `~3` (which was used to designate "stable releases').